subprojects/docs/src/snippets/developingPlugins/reactingToBuildFeatures/groovy/buildSrc/src/main/java/MyPlugin.java

```diff
@@ -0,0 +1,43 @@
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.configuration.BuildFeatures;
+
+import javax.inject.Inject;
+
+// tag::my-plugin[]
+public abstract class MyPlugin implements Plugin<Project> {
+
+    private final MyReport myReport = new MyReport();
```

Comment:
☁️ I'd remove this field and move the report instance to the `apply` method so the `BuildFeatures` injection is the first and only thing the user sees at this level

Reply:
Great suggestion!